plugins/systemd/systemd_mem

```diff
@@ -51,28 +58,51 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SO
 
 . "$MUNIN_LIBDIR/plugins/plugin.sh"
 
+# at least on debian stretch services report this as unlimited / unknown value
+SYSTEMD_UNLIMITED="18446744073709551615"
+
+for_services() {
+    if [ "$all_services" != "" ]; then
```

Comment:
As a matter of *declaring the interface*, I would suggest to add the following line somewhere at the beginning of the script:
```shell
all_services=${all_services:-}
```

This allows you to use `set -u` and it "declares" prominently, that the plugin is using this environment variable (interpreting it as an empty string, if unset).

The same would probably be good for `services`, too.

Reply:
thanks for the hint. did that for `all_services` and `services`